crates/trigger/src/runtime_config.rs

```diff
@@ -0,0 +1,262 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Result};
+use serde::Deserialize;
+use spin_config::provider::{env::EnvProvider, vault::VaultProvider};
+use toml;
+
+pub const DEFAULT_STATE_DIR: &str = ".spin";
+const DEFAULT_LOGS_DIR: &str = "logs";
+
+const SPIN_CONFIG_ENV_PREFIX: &str = "SPIN_CONFIG";
+
+const DEFAULT_SQLITE_DB_FILENAME: &str = "sqlite_key_value.db";
+
+/// RuntimeConfig allows multiple sources of runtime configuration to be
+/// queried uniformly.
+#[derive(Debug, Default)]
+pub struct RuntimeConfig {
+    local_app_dir: Option<PathBuf>,
+    files: Vec<RuntimeConfigOpts>,
+    overrides: RuntimeConfigOpts,
+}
+
+impl RuntimeConfig {
+    // Gives more consistent conditional branches
+    #![allow(clippy::manual_map)]
+
+    pub fn new(local_app_dir: Option<PathBuf>) -> Self {
+        Self {
+            local_app_dir: local_app_dir.map(Into::into),
+            ..Default::default()
+        }
+    }
+
+    /// Load a runtime config file from the given path. Options specified in a
+    /// later-loaded file take precedence over any earlier-loaded files.
+    pub fn load_config_file(&mut self, path: impl AsRef<Path>) -> Result<()> {
```

Comment:
If there can be multiple files in play, then maybe `merge_config_from` rather than `load`???